node/bindnode/infer.go

```diff
@@ -87,6 +104,78 @@ func fieldNameFromSchema(name string) string {
 	return strings.Title(name)
 }
 
+var defaultTypeSystem schema.TypeSystem
+
+func init() {
+	defaultTypeSystem.Init()
+
+	defaultTypeSystem.Accumulate(schemaTypeBool)
+	defaultTypeSystem.Accumulate(schemaTypeInt)
+	defaultTypeSystem.Accumulate(schemaTypeFloat)
+	defaultTypeSystem.Accumulate(schemaTypeString)
+	defaultTypeSystem.Accumulate(schemaTypeBytes)
+	defaultTypeSystem.Accumulate(schemaTypeLink)
+}
+
+// TODO: support IPLD maps and unions in inferSchema
+
+// TODO: support bringing your own TypeSystem?
+
+// TODO: we should probably avoid re-spawning the same types if the TypeSystem
+// has them, and test that that works as expected
```

Comment:
this seems like an optimization, right? keeping multiple primitive types reflecting each distinct aliased name in go seems like a reasonable place to end up.

Reply:
On one hand, yes, it's about deduplication. If I call Wrap on the same Go type ten thousand times, I don't want to add ten thousand schema types to a typesystem. They should all use the same schema type pointer.

On the other hand, it's also about correctness. If you call Wrap on two values of the same Go type, which gives you a TypedNode, and you grab their schema types... I think one would expect the two to have exactly the same schema type pointer.